### PR TITLE
Delete a few unused definitions from coreclr pal

### PR DESCRIPTION
**Pull request by am11**
Some of them were never used, some were rendered unused in recent code refactoring.

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]
Thank you!

**Comment by am11** [comment]
@AaronRobinsonMSFT, I was looking at the single usage of WszSearchPath (alias of SearchPathWrapper, which uses PAL's SearchPathW). It is used in the handling of P_INCLUDE control directives defined here: https://github.com/dotnet/runtime/blob/af2a48ddaec9a3e27002b0d3b80c07431c2e38c1/src/coreclr/inc/il_kywd.h#L289-L296 Are these all valid in some flavor of IL? (#ifndef and #include look like they shouldn't be there)

**Comment by AaronRobinsonMSFT** [comment]
> Are these all valid in some flavor of IL? (#ifdef, #ifndef and #include look like they shouldn't be there)

@am11 Yes, those are all valid keywords for the ILAsm toolchain. They are used in our testing in some places. For example:

https://github.com/dotnet/runtime/blob/af2a48ddaec9a3e27002b0d3b80c07431c2e38c1/src/tests/JIT/Directed/debugging/debuginfo/tests.il#L17-L23

https://github.com/dotnet/runtime/blob/af2a48ddaec9a3e27002b0d3b80c07431c2e38c1/src/tests/JIT/IL_Conformance/Old/Conformance_Base/bgt_un_r4.il#L6-L14

**Comment by am11** [comment]
Makes sense. `ilasm -include <mypath>` then `#include "somefile.il"` in code will search for `somefile.il`  in `mypath`.

File: src/coreclr/pal/inc/pal.h [hunk 1]
```diff
@@ -3452,7 +3437,6 @@ InterlockedDecrement(
     return result;
 }
 
-#define InterlockedDecrementAcquire InterlockedDecrement
```
Comment (jkotas) [inline src/coreclr/pal/inc/pal.h:3455]: I would keep `InterlockedDecrementAcquire`. It mirrors `InterlockedDecrementRelease` and so it is odd to have one without the other.

**Review by jkotas — APPROVED** [review]
Thanks

File: src/coreclr/pal/inc/pal.h (hunk 1 above)
Comment (am11) [reply src/coreclr/pal/inc/pal.h:3455]: Sure, I will add it back. Should `InterlockedDecrementAcquire` use `InterlockedIncrement` or was decrement correct method?

**Comment by jkotas** [comment]
> SearchPathW

It would be fine to move the implementation to ilasm, trim it to just what ilasm needs and share the trimmed implementation across all OSes.

File: src/coreclr/pal/inc/pal.h (hunk 1 above)
Comment (jkotas) [reply src/coreclr/pal/inc/pal.h:3455]: Actually, after looking at what all we have here - it is fine to delete it. We do not have a lot of consistency in what we have here. No action needed.

> Should InterlockedDecrementAcquire use InterlockedIncrement or was decrement correct method?

decrement was correct method.

**Comment by am11** [comment]
Thanks, I am moving SearchPathW to ilasm where it is used.

**Pull request merged**